packages/data/src/registry.js

```diff
@@ -278,19 +139,108 @@ export function createRegistry( storeConfigs = {} ) {
+		const namespace = createNamespace( reducerKey, { reducer }, registry );
+		registerGenericStore( reducerKey, namespace );
+		return namespace.store;
+	};
+
+	/**
+	 * Registers actions for external usage.
+	 *
+	 * @param {string} reducerKey   Part of the state shape to register the
+	 *                              selectors for.
+	 * @param {Object} actions   Actions to register.
+	 */
+	registry.registerActions = ( reducerKey, actions ) => {
+		const namespace = createNamespace( reducerKey, { actions }, registry );
+		registerGenericStore( reducerKey, namespace );
+	};
+
+	/**
+	 * Registers selectors for external usage.
+	 *
+	 * @param {string} reducerKey   Part of the state shape to register the
+	 *                              selectors for.
+	 * @param {Object} selectors    Selectors to register. Keys will be used as the
+	 *                              public facing API. Selectors will get passed the
+	 *                              state as first argument.
+	 */
+	registry.registerSelectors = ( reducerKey, selectors ) => {
+		const namespace = createNamespace( reducerKey, { selectors }, registry );
+		registerGenericStore( reducerKey, namespace );
+	};
+
+	/**
+	 * Registers resolvers for a given reducer key. Resolvers are side effects
+	 * invoked once per argument set of a given selector call, used in ensuring
+	 * that the data needs for the selector are satisfied.
+	 *
+	 * @param {string} reducerKey   Part of the state shape to register the
+	 *                              resolvers for.
+	 * @param {Object} resolvers Resolvers to register.
+	 */
+	registry.registerResolvers = ( reducerKey, resolvers ) => {
+		const namespace = createNamespace( reducerKey, { resolvers }, registry );
+		registerGenericStore( reducerKey, namespace );
+	};
+
+	/**
+	 * Registers a standard `@wordpress/data` store.
+	 *
+	 * @param {string} reducerKey Reducer key.
+	 * @param {Object} options    Store description (reducer, actions, selectors, resolvers).
+	 *
+	 * @return {Object} Registered store object.
+	 */
+	registry.registerStore = ( reducerKey, options ) => {
+		if ( ! options.reducer ) {
+			throw new TypeError( 'Must specify store reducer' );
+		}
+
+		const namespace = createNamespace( reducerKey, options, registry );
+		registerGenericStore( reducerKey, namespace );
+		return namespace.store;
+	};
+
+	/**
+	 * Registers a generic store.
+	 *
+	 * @param {string} key    Store registry key.
+	 * @param {Object} config Configuration (getSelectors, getActions, subscribe).
+	 */
+	function registerGenericStore( key, config ) {
+		if ( typeof config.getSelectors !== 'function' ) {
+			throw new TypeError( 'config.getSelectors must be a function' );
+		}
+		if ( typeof config.getActions !== 'function' ) {
+			throw new TypeError( 'config.getActions must be a function' );
+		}
+		if ( ! config.subscribe ) {
+			throw new TypeError( 'config.subscribe must be a function' );
+		}
+		stores[ key ] = config;
+		config.subscribe( globalListener );
```

Comment:
Having the custom store be responsible for tracking its own subscribers seems like a non-trivial task. I wonder if this should be inverted, where the custom store invokes a callback provided by the data module to be called when its own internal state is known to have been updated (callback as return value of `registerGenericStore`?).

Reply:
That's an interesting idea. I can try it out.

Reply:
So, I tried this out a bit, and I'm not sure that I like the idea of just assuming that the code which called `registerGenericStore` hooks up the event handler correctly. I feel that the `subscribe` method is a better contract. 

I did add a test for this, and made the simplest subscribe function I could. Since we can assume only one listener for this application, and no need to unsubscribe, it can be very simple:
```
	let listener = () => {};
	const storeChanged = () => {
		listener();
	};
	subscribe = ( newListener ) => {
		listener = newListener;
	};
```